### PR TITLE
Clef space to thick bar line

### PR DESCRIPTION
**Pull request by DavidBauer1984**
This PR keeps some space between system clefs and thick bar lines (rendered via `--bar-line-width`).

| Before | After |
| ------ | ----- |
| Before | After |

### CODE REVIEW

**Comment by lpugin** [comment]
That looks good. However, as I understand the issue is only with the scoreDef line and not all bar lines. So I wonder if we should not instead fix https://github.com/rism-digital/verovio/issues/2676 and at the same time place the system scoreDef line on the left of the system? That way we would not need to take into account its width for the placement of the clef.

image

**Comment by rettinghaus** [comment]
I'm not sure I see the connection of this fix with #2676.

**Comment by lpugin** [comment]
> I'm not sure I see the connection of this fix with https://github.com/rism-digital/verovio/issues/2676.

What I am trying to say it that instead of having the scoreDef line overlapping with the staff lines we should have it placed on the left of the staff lines as illustrated with the image above. Then we would not have to adjust the clef position depending on the presence of a line in the scoreDef and of its width, which in my opinion would be better.

If we change https://github.com/rism-digital/verovio/blob/3749e205a6249bf28f6e6c5ef229b08194d49efa/src/view_page.cpp#L363 to:
```cpp
this->DrawVerticalLine(dc, yTop, yBottom, x - barLineWidth / 2, barLineWidth);
```
Then rendering changes from
image
to
image

Obviously the second one is not perfect because the height of the bar does not take into account the staff line width and the bar then overlaps with the brace. This would still need to be fixed.

The reason this relates to #2676 is that if we do this, the line is actually drawn in the margin and will not show up if the margin is set to 0:

image

**Comment by rettinghaus** [comment]
Ok, thanks for the explanation. But then we would have different behaviors for the left and right (bar) lines of each system, right? I'm not sure this would be a good approach.

**Comment by lpugin** [comment]
ScoreDef lines are not bar lines. If the previous system had no bar line at the end you would still draw the scoreDef line. This is actually the left barline of the first measure
image
So I don't see the problem of treating them differently. 

We can very easily adjust the position of the first measure in `System::AlignMeasures` to make sure that the score Def line is shown if the margin is 0. 

Anyway, it does not really matter because. I'll review it further.

File: src/measure.cpp [hunk 1]
```diff
@@ -1047,6 +1059,11 @@ int Measure::AdjustXPos(FunctorParams *functorParams)
         StaffAlignment *staffAlignment = system->m_systemAligner.GetStaffAlignmentForStaffN(staffN);
         params->m_staffSize = (staffAlignment) ? staffAlignment->GetStaffSize() : 100;
 
+        // Prevent collisions of scoredef clefs with thick barlines
+        if (this->IsFirstInSystem() && system->GetDrawingScoreDef()->HasSystemStartLine()) {
+            params->m_upcomingMinPos = params->m_doc->GetDrawingBarLineWidth(params->m_staffSize);
+        }
```
Comment (lpugin) [inline src/measure.cpp:1065]: You do now want to check if the measure is the first of the system and if the scoreDef has a staff line for every staff in the measure. That will be quite inefficient for an orchestral score. Move this outside the loop and use a variable for resetting `m_upcomingMinPos`

File: src/view_page.cpp [hunk 2]
```diff
@@ -353,14 +353,14 @@ void View::DrawStaffGrp(
...
-            || (m_doc->GetCurrentScoreDef()->GetSystemLeftline() == BOOLEAN_true))) {
-        // int barLineWidth = m_doc->GetDrawingElementDefaultSize("bracketThickness", staffSize);
-        const int barLineWidth = m_doc->GetDrawingBarLineWidth(staffSize);
-        this->DrawVerticalLine(dc, yTop, yBottom, x + barLineWidth / 2, barLineWidth);
+    if (topStaffGrp) {
+        ScoreDef *scoreDef = vrv_cast<ScoreDef *>(staffGrp->GetFirstAncestor(SCOREDEF));
+        if (scoreDef && scoreDef->HasSystemStartLine()) {
+            const int barLineWidth = m_doc->GetDrawingBarLineWidth(staffSize);
+            this->DrawVerticalLine(dc, yTop, yBottom, x + barLineWidth / 2, barLineWidth);
+        }
```
Comment (lpugin) [inline src/view_page.cpp:361]: Since the parent scoreDef is used below in the function move it up and use the same local variable

**Pull request merged**